gameoflife.py

```diff
@@ -19,51 +18,71 @@
                                                               
                                                               '''
 
-def init_board(file_path):
-    b = []
+def init_board(file_path : str) ->list:
+    '''
+    Function initializes a board sccording to the content in file.
+    Input: file path.
+    Output: initialized board.
+    '''
+    board = []
     if not os.path.isfile(file_path):
         raise Exception(f"File {file_path} could not be found.")
     with open(file_path, mode="r") as file:
         file_content = file.readlines()
     for line in file_content:
-        b_line = []
+        board_line = []
         for cell in line:
             if cell == ALIVE_FORMAT:
-                b_line.append(ALIVE)
+                board_line.append(ALIVE)
             elif cell == DEAD_FORMAT:
-                b_line.append(DEAD)
+                board_line.append(DEAD)
             elif cell != ',' and cell !='\n':
                 raise Exception(f"File {file_path} is not written according format.")
-        b.append(b_line)
-    return b
+        board.append(board_line)
+    return board
 
 
-def sum_neighbors(row, col, board):
+def sum_neighbors(row : int, col : int, board : list) ->int:
+    '''
+    Function counts the alive neighbors around the wanted cell in the board.
+    Input: the wanted call location in board, board.
+    Output: the number of alive cells around the wanted cell.
+    '''
     
     neighbors_count = 0
     for i in range(-1,2):
         for j in range(-1,2):
             if i or j: # Don't count current cell 
-                if(0 <= row+i < len(board) and 0 <= col+j < len(board[row+i]) and board[row+i][col+j]):
-                    neighbors_count += 1
+                if 0 <= row+i < len(board) and 0 <= col+j < len(board[row+i]):
+                    neighbors_count += board[row+i][col+j]
     
     return neighbors_count
 
 
-def advance_generation(board):
+def advance_generation(board : list) ->list:
```

Comment:
you can specify the type of the items in the list, like this:
```
list[str]
list[int]
list[list[int]]
list[dict[str, int]]
```

please do it where needed in all the program